### PR TITLE
Add method/scope to identify participants eligible to change cohort and continue training

### PR DESCRIPTION
**Pull request by ethax-ross**
### Context

[Jira-3060](https://dfedigital.atlassian.net.mcas.ms/jira/software/projects/CPDLP/boards/87?assignee=712020%3A5b5839dc-e4d6-4df5-9dd6-627cc49fe345&selectedIssue=CPDLP-3060)

As part of the work to migrate 2021 candidates to the 2024 cohort to continue training we want to add methods on `ParticipantProfile` so that we can identify them and a scope so that we can find them all.

### Changes proposed in this pull request

-  Add method to identify if participant can migrate cohort

Add a method `can_change_cohort_and_continue_training?` to check if a participant is eligible for changing cohort in order to continue training in a later cohort, even though they have paid/payable/eligible declarations.

- Add scope to retrieve all participants eligible to migrate cohort

Update the existing method to use the scope for simplicity.

### Guidance to review

The numbers appear consistent with Nathan's word doc (and I've checked with Nathan as well and he's happy):

```
ParticipantProfile::ECT.eligible_to_change_cohort_and_continue_training(in_cohort_start_year: 2024).count
3,199 (3,500 in the word doc)

ParticipantProfile::Mentor.eligible_to_change_cohort_and_continue_training(in_cohort_start_year: 2024).count
14,867 (15,000 in the word doc)
```

I'm not super if `eligible_to_change_cohort_and_continue_training` is the best way to describe this action/pot of participants; open to suggestions.

### CODE REVIEW

File: app/models/participant_profile.rb [hunk 1]
```diff
@@ -73,6 +73,20 @@ def approved?
   # delivery_partner
   delegate :delivery_partner, to: :latest_induction_record, allow_nil: true
 
+  def self.eligible_to_change_cohort_and_continue_training(in_cohort_start_year:)
+    billable_states = %w[eligible payable paid].freeze
```
Comment (cwrw) [inline app/models/participant_profile.rb:77]: not in the scope of this pr but we should have this in declarations as a constant similar to statement line items as outside the scope it's used everywhere and needs cleanup 😂

File: app/models/participant_profile.rb [hunk 2]
```diff
@@ -73,6 +73,20 @@ def approved?
...
+
+    joins(:participant_declarations, schedule: :cohort)
+      .where(cohorts: { start_year: in_cohort_start_year - Cohort::OPEN_COHORTS_COUNT })
+      .where("participant_declarations.state IN (?) AND declaration_type != ?", billable_states, "completed")
+      .where.not(id: ParticipantDeclaration.billable.for_declaration(:completed).pluck(:participant_profile_id))
```
Comment (cwrw) [inline app/models/participant_profile.rb:82]: Why do we need this as well as thought the previous query is excluding billable completed declaration profiles 🤔

**Review by cwrw — COMMENTED** [review]
very cleanly done ⭐ have a few questions but otherwise 💯

File: app/models/participant_profile.rb (hunk 1 above)
Comment (ethax-ross) [reply app/models/participant_profile.rb:77]: Yeah definitely, it would be nice to tidy all that up

File: app/models/participant_profile.rb (hunk 2 above)
Comment (ethax-ross) [reply app/models/participant_profile.rb:82]: Yeah its a bit hard to follow, but essentially we want profiles with billable declarations that are not completed and also have 0 billable declarations that are completed

**Review by cwrw — APPROVED** [review]
⭐
